File: 0278-first-bad-version/0278-first-bad-version.c
// The API isBadVersion is defined for you.
// bool isBadVersion(int version);

int firstBadVersion(int n) {
    int l = 1, r = n;
    while(l <= r) 
    {
        int m = l + (r - l) / 2;
        if(isBadVersion(m))
        {
            r = m - 1;
        }
        else
        {
            l = m + 1;
        }
    }
    return l;
} 

File: 0278-first-bad-version/0278-first-bad-version.py
# The isBadVersion API is already defined for you.
# def isBadVersion(version: int) -> bool:

class Solution:
    def firstBadVersion(self, n: int) -> int:
        l, r = 1, n
        while l < r:
            m = (l + r) // 2
            if r - l >= 2:
                if isBadVersion(m):
                    r = m
                else:
                    l = m
            else:
                if isBadVersion(m):
                    return m
                else:
                    return m + 1
        return l 